Cython/Compiler/Code.py

```diff
@@ -2089,14 +2089,14 @@ def put_temp_declarations(self, func_context):
                 self.putln("%s%s;" % (static and "static " or "", decl))
 
         if func_context.should_declare_error_indicator:
-            if self.funcstate.uses_error_indicator:
-                unused = ''
-            else:
-                unused = 'CYTHON_UNUSED '
             # Initialize these variables to silence compiler warnings
-            self.putln("%sint %s = 0;" % (unused, Naming.lineno_cname))
-            self.putln("%sconst char *%s = NULL;" % (unused, Naming.filename_cname))
-            self.putln("%sint %s = 0;" % (unused, Naming.clineno_cname))
+            self.putln("int %s = 0;" % Naming.lineno_cname)
+            self.putln("const char *%s = NULL;" % Naming.filename_cname)
+            self.putln("int %s = 0;" % Naming.clineno_cname)
+            if not self.funcstate.uses_error_indicator:
+                self.putln("CYTHON_UNUSED_VAR(%s);" % Naming.lineno_cname)
+                self.putln("CYTHON_UNUSED_VAR(%s);" % Naming.filename_cname)
+                self.putln("CYTHON_UNUSED_VAR(%s);" % Naming.clineno_cname)
```

Comment:
probably same

Reply:
You might use `code.insertion_point` to create a sub-writer for all the declarations (thus ensuring they end up first)